polkadot/node/core/approval-voting/src/approval_db/v2/mod.rs

```diff
@@ -241,12 +254,25 @@ pub struct BlockEntry {
 	// block. The block can be considered approved if the bitfield has all bits set to `true`.
 	pub approved_bitfield: Bitfield,
 	pub children: Vec<Hash>,
+	// A list of candidates that has been approved, but we didn't not sign and
+	// advertise the vote yet.
+	pub candidates_pending_signature: BTreeMap<CandidateIndex, CandidateSigningContext>,
```

Comment:
Actually, we do need them to be sorted by CandidateIndex when we generate the signature, because signing `CandidateHashOfIndex1, CandidateHashOfIndex2` is different from signing `CandidateHashOfIndex2, CandidateHashOfIndex1`, so we would need to keep the hashes either sorted by the CandidateIndex or sort them at the moment we generate the signature in the case I use a VecDeque.